test/presets/netlify.test.ts

```diff
@@ -137,3 +187,139 @@ describe("nitro:preset:netlify", async () => {
     }
   );
 });
+
+describe("nitro:preset:netlify-edge", async () => {
+  const publicDir = resolve(getPresetTmpDir("netlify-edge"), "dist");
+  const ctx = await setupTest("netlify-edge", {
+    config: {
+      framework: {
+        name: "mock-framework",
+        version: "1.2.3",
+      },
+      publicAssets: [
+        {
+          fallthrough: true,
+          baseURL: "foo",
+          dir: publicDir,
+        },
+        {
+          fallthrough: false,
+          dir: publicDir,
+        },
+        {
+          fallthrough: true,
+          dir: publicDir,
+        },
+        {
+          baseURL: "icons",
+          dir: publicDir,
+        },
+        {
+          fallthrough: false,
+          baseURL: "nested/fonts",
+          dir: publicDir,
+        },
+      ],
+      output: {
+        publicDir,
+      },
+      netlify: {
+        images: {
+          remote_images: ["https://example.com/.*"],
+        },
+      },
+    },
+  });
+  testNitro(
+    ctx,
+    async () => {
+      const { default: handler } = (await import(
+        resolve(ctx.rootDir, ".netlify/edge-functions/server/server.js")
+      )) as {
+        default: (req: Request, _ctx: EdgeFunctionContext) => Promise<Response>;
+      };
+      return async ({ url: rawRelativeUrl, headers, method, body }) => {
+        // creating new URL object to parse query easier
+        const url = new URL(`https://example.com${rawRelativeUrl}`);
+        const req = new Request(url, {
+          headers: headers ?? {},
+          method,
+          body,
+        });
+        const res = await handler(req, {} as EdgeFunctionContext);
+        if (!(res instanceof Response))
+          // The Netlify Edge Function handler API allows returning `undefined` but this
+          // test helper only supports a Response or this shape.
+          return {
+            data: undefined,
+            status: 404,
+            headers: {},
+          };
+        return res;
```

Comment:
@pi0 I've got one test failure from the `testNitro()` cases that I don't understand:

```
 FAIL  test/presets/netlify.test.ts > nitro:preset:netlify-edge > API Works
AssertionError: expected undefined to match object { message: 'Hello API' }

- Expected:
Object {
  "message": "Hello API",
}

+ Received:
undefined

 ❯ test/tests.ts:229:26
    227|   it("API Works", async () => {
    228|     const { data: helloData } = await callHandler({ url: "/api/hello" });
    229|     expect(helloData).to.toMatchObject({ message: "Hello API" });
       |                          ^
    230|
    231|     if (ctx.nitro?.options.serveStatic) {
```

It's hitting this case where the edge function handler is returning `undefined`. Any tips? I'm not sure what's being tested here and why it would be behaving differently for netlify-edge.

Reply:
`/api/hello` is expected to be prerendered (an static asset in output public dir)